bsonutil/indexes.go

```diff
@@ -103,6 +103,28 @@ func ConvertLegacyIndexOptions(indexOptions bson.M) {
 	}
 }
 
+// ConvertLegacyIndexOptions removes options that don't match a known list of index options.
```

Comment:
This should say `ConvertLegacyIndexOptionsFromOp`. Also would be nice to add something saying that this is for oplog objects to differentiate it from the `ConvertLegacyIndexOptions` comment.